fix(GRAPHQL): Add error handling for unrecognized args to generate directive.

anurags92 opened the pull request:
DGRAPH PR Message:



### Motivation

1. Added error handling in gqlparser (https://github.com/dgraph-io/gqlparser/pull/16) and bumped the version of gqlparser used in dgraph. 
2. Removed stray printing of typeName which was polluting the logs

This PR fixes: GRAPHQL 1076, 1122


### Components affected by this PR <!-- (Optional) -->
GQL Parser






### Does this PR break backwards compatibility?

No


Fixes 
<!--
Your title must be in the following format: topic(Area): Feature
Topic must be one of build|ci|docs|feat|fix|perf|refactor|chore|test

Sample Titles:
feat(Enterprise): Backups can now get credentials from IAM
fix(Query): Skipping floats that cannot be Marshalled in JSON
perf: [Breaking] json encoding is now 35% faster if SIMD is present
chore: all chores/tests will be excluded from the CHANGELOG

Please add a description with these things:
1. A good description explaining the problem and what you changed.
2. If it fixes any GitHub issues, say "Fixes #GitHubIssue".
3. If it corresponds to a Jira issue, say "Fixes DGRAPH-###".
4. If this is a breaking change, please put "[Breaking]" in the title. In the description, please put a note with exactly who these changes are breaking for.
-->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dgraph-io/dgraph/7612)
<!-- Reviewable:end -->

vmrajas reviewed (commented):
Can you add a small error test in graphql/schema/gqlschema_test.yml ?

anurags92 commented:
> Can you add a small error test in graphql/schema/gqlschema_test.yml ?

Made[ changes to the gqlparser ](https://github.com/dgraph-io/gqlparser/pull/16)and added the tests in that repo. Updated this PR to reflect only version change.
